migrate/src/main/scala/migrate/ScalaMigrat.scala

```diff
@@ -30,13 +30,23 @@ class ScalaMigrat(scalafixSrv: ScalafixService) {
     for {
       compiler               <- setupScala3Compiler(scala3Classpath, scala3ClassDirectory, scala3CompilerOptions)
       (scalaFiles, javaFiles) = unmanagedSources.partition(_.value.endsWith("scala"))
-      initialFileToMigrate <-
-        buildMigrationFiles(scalaFiles)
-      _ <- compileInScala3(initialFileToMigrate, javaFiles ++ managedSources, compiler)
-      initialFileToMigrate <-
-        buildMigrationFiles(unmanagedSources)
-      _            <- compileInScala3(initialFileToMigrate, managedSources, compiler)
-      migratedFiles = initialFileToMigrate.map(f => (f.source, f.migrate(compiler))).toMap
+      // first try to compile without adding any patch
+      filesWithErr = compileInScala3AndGetFilesWithErrors(unmanagedSources ++ managedSources, compiler)
+      migratedFiles <- if (filesWithErr.isEmpty) {
+                         scribe.info("The project compiles successfully in Scala 3")
+                         Success(Map[AbsolutePath, FileMigrationState.FinalState]())
+                       } else {
+                         val filesWithoutErrors = scalaFiles.diff(filesWithErr)
+                         for {
+                           initialFileToMigrate <- buildMigrationFiles(filesWithErr)
+                           _ <- compileInScala3(
+                                  initialFileToMigrate,
+                                  filesWithoutErrors ++ javaFiles ++ managedSources,
+                                  compiler
+                                )
```

Comment:
This should fix the compilation errors in the `filesWithErr` but there may be cases where the compiler then find errors in others files, in which case we can loop to add the patches of the new `filesWithErr` until there is no more error.

Can be done in a V2 after we have found a case in which it happens.

Reply:
Good idea. I have to open an issue to get track of this.